internal/repository/index.go

```diff
@@ -210,21 +210,34 @@ func (idx *Index) indexEntryToPackedBlob(h restic.BlobHandle, entry indexEntry)
 }
 
 // Lookup queries the index for the blob ID and returns a restic.PackedBlob.
-func (idx *Index) Lookup(id restic.ID, tpe restic.BlobType) (blobs []restic.PackedBlob, found bool) {
+func (idx *Index) Lookup(id restic.ID, tpe restic.BlobType) (blob restic.PackedBlob, found bool) {
 	idx.m.Lock()
 	defer idx.m.Unlock()
 
 	h := restic.BlobHandle{ID: id, Type: tpe}
 
-	blob, ok := idx.blob[h]
+	entry, ok := idx.blob[h]
 	if ok {
-		blobList := idx.withDuplicates(h, blob)
-		blobs = make([]restic.PackedBlob, 0, len(blobList))
+		return idx.indexEntryToPackedBlob(h, entry), true
+	}
 
-		for _, p := range blobList {
-			blobs = append(blobs, idx.indexEntryToPackedBlob(h, p))
-		}
+	return restic.PackedBlob{}, false
+}
+
+// LookupAll queries the index for the blob ID and returns all entries including
+// duplicates.
+func (idx *Index) LookupAll(id restic.ID, tpe restic.BlobType) (blobs []restic.PackedBlob, found bool) {
```

Comment:
Here again, the second return value isn't necessary. The caller can just check the len of the slice.

Reply:
You are right, again :smiley: 
I changed this, thanks for the suggestion!